lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -265,6 +276,63 @@ public void run() {
 		}
 	}
 
+	/**
+	 * Authentication token changes for the current connection are possible when
+	 * the client is in the {@link ConnectionState#connected}, {@link ConnectionState#connecting}
+	 * or {@link ConnectionState#disconnected } states.
+	 *
+	 * In the current protocol version we are not able to update auth params on the fly;
+	 * so disconnect, and the new auth params will be used for subsequent reconnection
+	 * * <p>
+	 * Spec: RTC8 (reauthorise)
+	 * </p>
+	 *
+	 */
+	public void onAuthUpdated() {
+		switch (state.state){
+			case connected:
+			case connecting:
+			case disconnected:
+				connection.once(ConnectionState.failed, new ConnectionStateListener() {
```

Comment:
If onAuthUpdated() invoked, I must do reconnect. In ConnectionManager, I use method requestState(ConnectionState state), it invokes close(),abort(),connect() methods for the Transport. So, I invoke disconnect(), lock the thread, and listen when connection state is changed to failed, than notify and do next step (connect). For connect used the same logic. In the end, onAuthUpdated() makes complete cycle of reconnect.